backend/src/db/migrations/1579387929111-merge_duplicate_location_nodes.js

```diff
@@ -0,0 +1,77 @@
+import { throwError, concat } from 'rxjs'
+import { flatMap, mergeMap, map, catchError } from 'rxjs/operators'
+import { getDriver } from '../neo4j'
+
+export const description = `
+  This migration merges duplicate :Location nodes. It became
+  necessary after we realized that we had not set up constraints for Location.id in production.
+`
+export function up(next) {
+  const driver = getDriver()
+  const rxSession = driver.rxSession()
+  rxSession
+    .beginTransaction()
+    .pipe(
+      flatMap(transaction =>
+        concat(
+          transaction
+            .run(
+              `
+                    MATCH (location:Location)
+                    RETURN location {.id}
+                  `,
+            )
+            .records()
+            .pipe(
+              map(record => {
+                const { id: locationIds } = record.get('location')
+                return { locationIds }
+              }),
+              mergeMap(({ locationIds }) => {
+                return transaction
+                  .run(
+                    ` 
+                    MATCH(location:Location {id: $locationIds}), (location2:Location {id: $locationIds})
+                    WHERE location.id = location2.id AND id(location) < id(location2)
+                    CALL apoc.refactor.mergeNodes([location, location2], { properties: 'combine', mergeRels: true }) YIELD node as updatedLocation
```

Comment:
This would create duplicate relationships between the merged node and the nodes which were connected to the previous nodes.

Reply:
would it?
so it should be false?